tests/rados/test_election_strategies.py

```diff
@@ -222,3 +283,382 @@ def check_mon_status(rados_obj, mon_obj):
+            > Mon becomes leader
+            > Mon is part of Quorum
+        - Scenario 3: connectivity -> disallow
+            > Mon does not become leader
+            > Mon is part of Quorum
+            > Mon is still in disallowed leaders list
+        - Switch back to previous election strategy
+        - Check Mon is not leader, Mon is in quorum and Mon is in disallowed leaders list
+        - Remove lowest ranked Mon from disallowed leaders list
+        - If original strategy was disallow, Check lowest ranked Mon becomes leader.
+          Check Mon is in Quorum and Mon is removed from disallowed leaders list
+
+    Args:
+        mon_obj: Object of MonElectionStrategies
+        election_strategy: disallow or connectivity
+    Example:
+        Scenario 1: disallow -> classic
+            check_disallowed_leaders_election_strategy(mon_obj, election_strategy="classic")
+        Scenario 2: connectivity -> classic
+            check_disallowed_leaders_election_strategy(mon_obj, election_strategy="classic")
+        Scenario 3: connectivity -> disallow
+            check_disallowed_leaders_election_strategy(mon_obj, election_strategy="disallow")
+    Returns:
+        None
+        Raises exception when tests fail
+    """
+    strategies = {1: "classic", 2: "disallow", 3: "connectivity"}
+    original_election_strategy = strategies[mon_obj.get_election_strategy()]
+
+    if (
+        original_election_strategy == 1
+        or original_election_strategy == election_strategy
+    ):
+        log.error(
+            "Cannot perform test in classic election strategy and"
+            " election strategy to switch cannot be same as current strategy"
+        )
+        return False
+
+    log.info(
+        "\n -------------------------------------------"
+        "\n Starting validation:"
+        f"\n Changing election strategy from {original_election_strategy} mode "
+        f"to {election_strategy} mode when lowest ranked Mon in disallowed leader list"
+        "\n -------------------------------------------"
+    )
+
+    lowest_ranked_mon = list(mon_obj.get_mon_quorum())[0]
+
+    log.info(f"Adding Mon {lowest_ranked_mon} to disallowed leaders list")
+    if not mon_obj.set_disallow_mon(mon=lowest_ranked_mon):
+        log.error(f"Could not add Mon {lowest_ranked_mon} to disallow list")
+        return False
+
+    log.info(f"Proceeding to perform checks on disallowed Mon {lowest_ranked_mon}")
+    # Check Mon is not leader, Mon is in quorum and Mon is in disallow leaders list
+    mon_obj.disallowed_leader_checks(lowest_ranked_mon)
+
+    log.info(f"Proceeding to set election strategy to {election_strategy}")
+    if not mon_obj.set_election_strategy(mode=election_strategy):
+        log.error(f"Could not set election strategy to {election_strategy} mode")
+        return False
+
+    log.info("Proceeding to checks on lowest ranked Mon as disallowed leader")
+    if election_strategy == "classic":
```

Comment:
use `ceph mon stat` instead of `ceph mon dump` to get quorum details.

Reply:
Added new function() which uses **ceph mon stat** to return same dictionary as previously used **ceph mon dump** function. 

https://github.com/red-hat-storage/cephci/blob/83e34286c4e52f3067a54ba0b8b151b8345e9bde/ceph/rados/monitor_workflows.py#L263-L274